rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -729,6 +643,239 @@ func (r *Layer2Relayer) handleL2RollupRelayerConfirmLoop(ctx context.Context) {
 	}
 }
 
+func (r *Layer2Relayer) constructCommitBatchPayload(dbBatch *orm.Batch) ([]byte, *kzg4844.Blob, error) {
+	if dbBatch.Index == 0 {
+		return nil, nil, fmt.Errorf("invalid args: batch index is 0, should only happen in committing genesis batch")
+	}
+
+	dbChunks, err := r.chunkOrm.GetChunksInRange(r.ctx, dbBatch.StartChunkIndex, dbBatch.EndChunkIndex)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch chunks: %w", err)
+	}
+
+	chunks := make([]*encoding.Chunk, len(dbChunks))
+	for i, c := range dbChunks {
+		blocks, getErr := r.l2BlockOrm.GetL2BlocksInRange(r.ctx, c.StartBlockNumber, c.EndBlockNumber)
+		if getErr != nil {
+			return nil, nil, fmt.Errorf("failed to fetch blocks: %w", getErr)
+		}
+		chunks[i] = &encoding.Chunk{Blocks: blocks}
+	}
+
+	parentDBBatch, getErr := r.batchOrm.GetBatchByIndex(r.ctx, dbBatch.Index-1)
+	if getErr != nil {
+		return nil, nil, fmt.Errorf("failed to get parent batch header: %w", getErr)
+	}
+
+	parentBatchHeader := parentDBBatch.BatchHeader
+	parentBatchHash := common.HexToHash(parentDBBatch.Hash)
+	startBlockNumber := dbChunks[0].StartBlockNumber
+	if startBlockNumber >= r.banachForkHeight { // codecv1
+		batch := &encoding.Batch{
+			Index:                      dbBatch.Index,
+			TotalL1MessagePoppedBefore: dbChunks[len(dbChunks)-1].TotalL1MessagesPoppedBefore + dbChunks[len(dbChunks)-1].TotalL1MessagesPoppedInChunk,
```

Comment:
I remember we used `parentBatch.TotalL1MessagePopped` before, did you change it so that we don't need to decode the parent batch?

Reply:
Rechecked. db batch does not contain `TotalL1MessagePopped`, so fetch this info through the db chunk as a walk-through, another approach is to decode the parent batch's encoding.